### PR TITLE
Bugfix geom_miss_point. Make lables factors

### PR DESCRIPTION
**Pull request by maksymiuks**
## Description

Fix error occurring when shape is passed in mapping. Basically when one tried to specify shape in aes passed to geom_miss_point (what technically should be possible) for some reason deafult_aes was used for colour that didn't have ..missing.. in env (nor parent env) resulting in error.

Additionally I introduced small change to how labels are constructed. Now they are factor instead of character so the Missing (and by extension Not Missing) will always have the same color regardless of input data. Previously have we split our data into two sub-data frames where one of them would have only complete cases, while second would also include NAs, then plot for the first sub-data frames appeared with different color mappings (on Missing and Not Missing) than second sub-data frame.

## Related Issue
https://github.com/njtierney/naniar/issues/290

## Example
Thanks to that code below will work

```
data <- DALEX::apartments
data <- missForest::prodNA(data)
data$no.rooms <- as.factor(data$no.rooms)

library(ggplot2)
library(naniar)

ggplot(data = data, mapping = aes(.data$m2.price, .data$surface)) +
  geom_miss_point()

ggplot(data = data, mapping = aes(.data$m2.price, .data$surface, color = .data$district)) +
  geom_miss_point()

ggplot(data = data, mapping = aes(.data$m2.price, .data$surface, shape = .data$district)) +
  geom_miss_point()

ggplot(data = data, mapping = aes(.data$m2.price, .data$surface, shape = .data$no.rooms, color = .data$district)) +
  geom_miss_point()
```

previously 3rd and 4th plot would not appear (error would be returned instead)

## Tests
I made changes to code that is already covered by unit tests. Addiitioanlly related tests were updated.

## NEWS + DESCRIPTION
Done

### CODE REVIEW

File: tests/testthat/test-label-miss-1d.R [hunk 1]
```diff
@@ -15,16 +15,16 @@ test_that("label_miss_1d returns a vector of the same length as the input",{
 
 test_that("label_miss_1d returns character vector",{
```
Comment (jonocarroll) [inline tests/testthat/test-label-miss-1d.R:16]: (suggestion) perhaps modify the description to reflect that this is a factor, and maybe a comment that `typeof(factor())` is "integer".

**Comment by jonocarroll** [comment]
Just to help with the review of this, here's my previous construction from #290 with these changes applied
``` r
library(naniar)
library(ggplot2)
mydat <- mtcars
mydat$mpg[1:10] = NA
ggplot(mydat, aes(x = mpg, y = disp, shape = factor(cyl))) +
  geom_miss_point(size = 4)
```



<sup>Created on 2021-11-30 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

FYI, I'm not sure if you need/want to keep the default colour order, but with these changes the Missing points are blue and the Not Missing points are red. That's a consequence of making Missing be the last entry in the factor. With all Not Missing this is consistent
``` r
mydat <- mtcars
ggplot(mydat, aes(x = mpg, y = disp, shape = factor(cyl))) +
  geom_miss_point(size = 4)
```



but with all Missing the points are mis-labelled Not Missing

``` r
mydat <- mtcars
mydat$mpg = NA
ggplot(mydat, aes(x = mpg, y = disp, shape = factor(cyl))) +
  geom_miss_point(size = 4)
```



<sup>Created on 2021-11-30 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

The reason here seems to be that passing all `NA` through `GeomPoint` results in a vector of non-missing values
```
[1] 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
attr(,"class")
[1] "mapped_discrete" "numeric"  
```
I can suggest an improvement to `label_miss_2d` of
```r
# all missing
if (all(x1 == 1) && inherits(x1, "mapped_discrete")) x1[] <- NA
if (all(x2 == 1) && inherits(x2, "mapped_discrete")) x2[] <- NA
temp <- data.frame(x1, x2) %>% is.na %>% rowSums() %>% factor(levels = c("0", "1", "2"))
forcats::fct_recode(temp, "Not Missing" = "0", "Missing" = "1", "Missing" = "2")
```
but this still results in only a single legend entry (albeit the correct one, now).

In all of these examples, as well as the 3rd DALEX one, the shape legend is broken (symbols not rendered).

Let us know if there's anything we can do to help reach a solution.

File: tests/testthat/test-label-miss-1d.R (hunk 1 above)
Comment (maksymiuks) [reply tests/testthat/test-label-miss-1d.R]: Great catch, I overlooked that description. Its fixed

**Comment by njtierney** [comment]
Thanks so much for this! This is excellent. I'll take a quick look soon at this locally but I'm sure it's all good.

**Comment by njtierney** [comment]
Hi @jonocarroll and @maksymiuks  - apologies this has taken me forever to look into this.

I'm a bit stuck on the fact that the colours are now swapped - I tried to change the factor order to the opposite, but this didn't work, but I'd really like to have this keep the colours the same as previous, I'm not quite sure how to manage this - do you have any thoughts?

**Comment by jonocarroll** [comment]
As far as I know (and that may be vastly out of date) if you want a colour to be stable you need to specify it with a manual scale. I'd suggest manually specifying colours for 'Missing' and 'Not Missing' regardless of the data, i.e. always having those legend entries present.

**Comment by njtierney** [comment]
OK I'll do that I reckon - that will also mean I can target https://github.com/njtierney/naniar/issues/148 at the same time, I've been meaning to set the colours to something colourblind safe!

**Comment by njtierney** [comment]
It is not as simple as I thought to set the colours manually via the custom geom - have you had experience doing this, @jonocarroll ?

**Comment by jonocarroll** [comment]
I haven't, sorry. I can try to reacquaint myself with this code sometime and see what I can do.

**Comment by njtierney** [comment]
Thanks! I'll try and have a stab as well, it's been a while since I've written a geom

**Comment by njtierney** [comment]
While I would prefer if the colours stayed the same, I think this solves an important issue so let's merge this once checks are done :)

**Comment by njtierney** [comment]
Resolves #290

**Comment by njtierney** [comment]
Aha! Figured it out - just needed to "revel" the factor levels

**Pull request merged**